### ISSUE TITLE
Grain interfaces with non-async static members fail with `ORLEANS0009`

### ISSUE THREAD

**Issue by minichma** [dotnet/orleans#8500]
Consider the following grain interface:

```cs
    public interface IFoo : IGrainWithStringKey
    {
        public static string Bar() => "bar";

        Task<string> Baz();
    }
```

For this interface the compiler issues a `ORLEANS0009` error, because the `Bar()` method is not async (unless the error is suppressed):

`error ORLEANS0009: Grain interfaces methods must return a compatible type, such as Task, Task<T>, ValueTask, ValueTask<T>, or void`

Because a static member cannot be invoked via a reference to a grain, this case should not be treated as an error.

Static interface methods could e.g. be helpful for composing or parsing of compound grain IDs.

**Comment by minichma** [comment]
Just saw, that this is a duplicate of #8389 and has already been addressed in v7.1.2. Sorry for the inconvenience.

**Comment by ReubenBond** [comment]
No worries, @minichma, and thank you for reporting anyway
